apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java

```diff
@@ -37,94 +38,126 @@
-          value -> claimBuilder.setLobCdEnumValue(Integer.parseInt(value)));
-      data.get(Fiss.ORG_NPI_NUM).ifPresent(claimBuilder::setNpiNumber);
-      data.get(Fiss.PRNCPAL_DGNS_CD).ifPresent(claimBuilder::setPrincipleDiag);
-      data.get(Fiss.PRVDR_NUM)
-          .ifPresent(value -> claimBuilder.setMedaProv6(String.format("%.6s", value)));
-
-      addDiagCodes(claimBuilder, data);
-      addProcCodes(claimBuilder, data);
-
-      return FissClaimChange.newBuilder()
-          .setTimestamp(Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()).build())
-          .setSeq(sequenceNumber.inc())
-          .setClaim(claimBuilder.build())
-          .setChangeType(ChangeType.CHANGE_TYPE_UPDATE)
-          .build();
+  public Optional<MessageOrBuilder> transform(
+      WrappedCounter sequenceNumber, Parser.Data<String> data, WrappedMessage message) {
+    FissClaimChange claimToReturn;
+
+    int lineNumber = getLineNumber(data, Fiss.CLM_LINE_NUM);
+
+    if (message.getMessage() instanceof FissClaimChange) {
+      // There is an existing claim from a previous run
+      FissClaimChange storedClaim = (FissClaimChange) message.getMessage();
+
+      if (message.getLineNumber() == (lineNumber - 1)) {
+        // If it's the next sequential line number, add to previous claim
+        message.setMessage(addToExistingClaim(storedClaim, data));
+        message.setLineNumber(lineNumber);
+        claimToReturn = null;
+      } else if (lineNumber == 1) {
+        // If the line number is 1, it's a new claim, return the old, store the new
+        claimToReturn = storedClaim;
+        message.setLineNumber(1);
+        message.setMessage(transformNewClaim(sequenceNumber, data));
+      } else {
+        // If it's not the next claim or a new one starting at 1, then something is wrong.
+        throw new IllegalStateException(
+            String.format(
+                "Invalid row sequence, previous line number: %d, current line number: %d",
+                message.getLineNumber(), lineNumber));
+      }
+    } else {
+      // This must be the first run, store a new claim
+      message.setLineNumber(1);
+      message.setMessage(transformNewClaim(sequenceNumber, data));
+      claimToReturn = null;
     }
 
-    // We're skipping any entry with CLM_LIN_NUM > 1
-    return null;
+    return Optional.ofNullable(claimToReturn);
+  }
+
+  @VisibleForTesting
+  FissClaimChange addToExistingClaim(FissClaimChange fissClaimChange, Parser.Data<String> data) {
+    FissClaim.Builder claimBuilder = fissClaimChange.getClaim().toBuilder();
+
+    // Not currently implemented for FISS claims, just store the same claim
+
+    return fissClaimChange.toBuilder().setClaim(claimBuilder.build()).build();
   }
 
   @VisibleForTesting
-  boolean isFirstLineNum(Parser.Data<String> data) {
-    Optional<String> lineNum = data.get(Fiss.CLM_LINE_NUM);
+  FissClaimChange transformNewClaim(WrappedCounter sequenceNumber, Parser.Data<String> data) {
```

Comment:
The various methods touched could use some documentation, particularly because they were renamed and partially rewritten.